openfl/federated/plan/plan.py

```diff
@@ -450,13 +483,101 @@ def get_data_loader(self, collaborator_name):
             {TEMPLATE: "openfl.federation.DataLoader", SETTINGS: {}},
         )
 
-        defaults[SETTINGS]["data_path"] = self.cols_data_paths[collaborator_name]
+        datasources = self._get_local_data_sources(collaborator_name, defaults)
+        datasources.extend(self._get_data_sources(collaborator_name))
+
+        if len(datasources) == 0:
+            raise ValueError(f"No data sources found for collaborator {collaborator_name}.")
+
+        vds = VerifiableDatasetInfo(
+            data_sources=datasources,
+            label="",
```

Comment:
What does `label` refer to? Couldn't infer from the class definition (also worth documenting)
It may cause confusion with `label` in the context of ML